cmd/libsnap-confine-private/utils.c

```diff
@@ -140,6 +140,38 @@ void write_string_to_file(const char *filepath, const char *buf)
 		die("fclose failed");
 }
 
+sc_identity sc_set_effective_identity(sc_identity identity)
+{
+	debug("set_effective_identity uid:%d (change: %s), gid:%d (change: %s)",
+	      identity.uid, identity.change_uid ? "yes" : "no",
+	      identity.gid, identity.change_gid ? "yes" : "no");
+	sc_identity old = {.change_gid = 0,.change_uid = 0 };
```

Comment:
I can explicitly memset the whole structure before handing out other assignments.

Reply:
> I can explicitly memset the whole structure before handing out other assignments.

I'm not opposed to that, but my point is more about algorithmically, this function is dependent on .change_uid and .change_gid being properly initialized/set since if, for example, an uninitialized .change_uid floated in, that conditional would evaluate to true when maybe we only meant for .change_gid to evaluate to true. A small terse comment that serves as a warning/reminder to a future code optimization/refactor/etc is all I'm looking for.